src/decorator/field-name.ts

```diff
@@ -0,0 +1,29 @@
+import { METADATA_DESERIALIZE_FIELD_NAME } from './deserialize-field-name';
+
+export const METADATA_CUSTOM_FIELDS = 'serializer:field:properties';
+/**
+ * Tags a property to map from a different field name.
+ *
+ * ## Example:
+ *
+ * ```typescript
+ * export class Example{
+ *      @FieldName('bar')
+ *      foo: string;
+ * }
+ *
+ * const result = serializer.deserialize<Example>({ bar: 'hey' }, Example);
+ * console.log(result.foo);
+ * ```
+ *
+ * This will print `hey` because the attribute with name `bar` has been mapped into the `foo` property.
+ *
+ * @decorator Property
+ */
+export function FieldName(fieldName: string): (...args: any[]) => void {
+    return (target: any, propertyKey: string) => {
+        const customFieldNames = Reflect.getMetadata(METADATA_CUSTOM_FIELDS, target) || [];
+        Reflect.defineMetadata(METADATA_CUSTOM_FIELDS, customFieldNames.concat([propertyKey]), target);
```

Comment:
TypeScript do the job. See the [example](https://www.typescriptlang.org/play/index.html#src=const%20test%20%3D%20%5B1%2C%202%2C%203%2C%204%2C%205%5D%3B%0D%0Aconst%20test2%20%3D%20%5B...test%2C%206%5D%3B)